python/test/unit/runtime/test_irsource.py

```diff
@@ -0,0 +1,93 @@
+import tempfile
```

Comment:
nit: maybe that should be in python/test/unit/tools/

Reply:
just moved [test_irsource.py](https://github.com/triton-lang/triton/pull/4924/files/873741fa6a37a1f9ccf45e642322f8f22220d205#diff-8d9acf64fe89456a443c88804ac2d9c7c8b85948ede2da0753e857e6a707b56b) to the tools dir